src/k8s/pkg/k8sd/features/coredns/status.go

```diff
@@ -0,0 +1,40 @@
+package coredns
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/canonical/k8s/pkg/snap"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CheckDNS checks the CoreDNS deployment in the cluster.
+func CheckDNS(ctx context.Context, snap snap.Snap) error {
```

Comment:
Same here. `CheckDNS` should only check once whether the expected pods are up and running